pyriemann/transfer/euclidean.py

```diff
@@ -0,0 +1,258 @@
+        return X_align
+
+    def _transform_cov(self, X, d):
+        """ Compute covariance matrices of trials in array X and save"""
+
+        cov = covariances(X, estimator=self.estimator)
+        self.cov_[d] = cov
+        return cov
+
+    def fit(self, X, y_enc):
+        """Fit (estimates) the reference matrix for each domain.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_channels) or (n_trials, n_channels, n_windows)
+            Data trials, can be SPD matrices or raw.
+        y_enc : ndarray, shape (n_trials,)
+            Extended labels for each trial. Encodes domain information.
+
+        Returns
+        -------
+        self : EuclideanAlignment instance
+            The EuclideanAlignment instance.
+        """
+
+        self.ref_.clear()
+
+        _, _, domains = decode_domains(X, y_enc)
+
+        self._fit_sw(X, domains)
+
+        return self
+
+    def transform(self, X, y_enc=None):
+        """Transforms target data points after calibration step.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_channels) or (n_trials, n_channels, n_windows)
+            Target trials, can be SPD matrices or raw.
+        y_enc : None
+            Not used, here for compatibility with sklearn API.
+
+        Returns
+        -------
+        X : ndarray, shape (n_trials, n_channels, n_channels) or (n_trials, n_channels, n_windows)
+            Target trials with Euclidean mean at Identity
+
+        """
+
+        # In this case, X is test and you suppose that calibration was previously fitted
+        ref = self.ref_[0]
+
+        if self.dtype == 'covmat':
+            X_align = ref @ X @ ref
+        else:
+            X_align = ref @ X
+
+        return X_align
+
+    def fit_transform(self, X, y_enc=None, **fit_params):
+        """Fit EuclideanAlignment and then transform data points.
+
+        Calculate the mean of all matrices in each domain and then recenter
+        them to Identity.
+
+        .. note::
+           This method is designed for using at training time. The output for
+           .fit_transform() will be different than using .fit() and
+           .transform() separately.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_channels) or (n_trials, n_channels, n_windows)
+            Target trials, can be SPD matrices or raw.
+        y_enc : ndarray, shape (n_matrices,)
+            Extended labels for each trial. Encodes domain information.
+
+        Returns
+        -------
+        X : ndarray, shape (n_trials, n_channels, n_channels) or (n_trials, n_channels, n_windows)
+            Target trials with Euclidean mean at Identity
+
+        """
+
+        self.fit(X, y_enc)
+
+        _, _, domains = decode_domains(X, y_enc)
+
+        X_align = self._transform_sw(X, domains)
+
+        return X_align
+
+    def __sklearn_is_fitted__(self):
```

Comment:
Your transformer always needs fitting. I don't think you can call transform without fit first. I suggest to remove this override.

Reply:
Yes, I agree! Supposing that you have data from different 'train' subjects (that will be used to train a classifier, for example) + calibration data from 'target' subject, and you want to use calibration data to compute reference for aligning the test data from 'target', the pipeline I was thinking was:

'Train' subjects --> `fit_transform` --> aligned 'Train'

'Calibration' --> `fit` calibration (compute reference)  or `fit_transform` if you want aligned calibration data

'Test' --> `transform` 'test' **after fitting calibration** --> aligned 'Test'

In this way, the last reference matrix computed/saved would be from calibration, and it would be used for transforming the test! I'm not sure if the way I implemented would be the best to do it, but that is what I was thinking